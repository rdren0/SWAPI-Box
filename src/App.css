
.App {
  text-align: center;
  display: grid;
  grid-template-rows: 100px 100px 300px;
}

body{
  background-image: url("https://images.pexels.com/photos/998641/pexels-photo-998641.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260");
  color: #ffd700;
  background-size: cover;
  background-repeat: no-repeat;
  margin: 0px;
}

.header {
  align-items: center;
  justify-content: center;
  background-image: none;
}

h1{
  font-size: 50px;
  margin-top: 10px;
  display: inline;
  color: white;
}

h3{
  margin-top: 5px;
  color: white;
}

p{
  overflow: hidden;
  text-align: justify;
  margin-left: 45px;
}

.card-area{
  height: 275px;
  width: 750px;
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  overflow: scroll;
  margin: auto;
}

.card{
  width: 180px;
  height: 260px;
  border: 2px solid white;
  margin: 10px;
  background-color: black;
}

button{
  border: 2px solid #ffd700;
  border-radius: 12px;
  color: white;
  font-size: 22px;
  width: 100px;
  background-color: black;
  margin: 3px;
}

button:hover{
  background-color: gainsboro;
  color: black;
}

#titles
{
  margin: auto;
  position: absolute;
  overflow: hidden;
  width: 800px;
  height: 75%;
  bottom: 0;
  font-size: 30px;
  font-weight: bold;
  transform-origin: 50% 100%;
  transform: perspective(300px) rotateX(25deg);
}

#titles:after{
  position: absolute;
  content: ' ';
  left: 0;
  right: 0;
  top: 0;
  bottom: 60%;
  background-image: linear-gradient(top, rgba(0,0,0,1) 0%, transparent 100%);
  pointer-events: none;
}

#titlecontent{
  position: absolute;
  top: 100%;
  animation: scroll 100s linear 4s infinite;
  padding: auto;
}

@keyframes scroll {
  0% { top: 100%; }
  100% { top: -170%; }
}

