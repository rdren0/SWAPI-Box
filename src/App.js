import React, {Component} from 'react';
import Header from './Components/Header/Header.js';
import Container from './Components/Container/Container.js';
import TextScroll from './Components/TextScroll/TextScroll.js';
import { fetchCall } from './fetchcall';
import PropTypes from 'prop-types';
import './App.css';



class App extends Component{
  constructor(){
    super()
    this.state = {
      cardData: [],
      currentCategory: '',
      randomMovie: {}
    }
  }

componentDidMount = () => {
  const url = "https://swapi.co/api/";
  fetchCall(url).then(results => this.gatherTypes(results.people))
  this.randomMovie("https://swapi.co/api/films")
}



gatherTypes = (results) => {
  fetchCall(results).then(results => this.setState({cardData: results.results}))
}


filterByType = (type) => {
  type = type.toLowerCase()
  const URL = `https://swapi.co/api/${type}/`
  this.setState({currentCategory: type})
  this.gatherTypes(URL);
}


randomMovie = (url) =>{
  const random = Math.floor(Math.random() * Math.floor(7));
  fetchCall(url).then(results => this.setState({randomMovie: results.results[random]}))
}


  render(){
    return (
      <div className="App">
      <Header className="header"/>
      <Container category = {this.state.currentCategory} cards = {this.state.cardData} filterByType = {this.filterByType} />
      <TextScroll crawl={this.state.randomMovie}/>
    </div>
    );
  }
}

App.propTypes = {
  cardData: PropTypes.array,
  randomMovie: PropTypes.object,
  currentCategory: PropTypes.string,
 
}

export default App;
