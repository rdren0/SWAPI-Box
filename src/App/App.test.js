import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from './App';


describe('App', () =>{
  const fetchSpy = jest.spyOn(global, 'fetch');
  const mockFunction = jest.fn()
  let wrapper;
  let URL;
  beforeEach(()=>{
    wrapper = shallow(<App 
      randomMovie={mockFunction} 
      filterByType={mockFunction}
      />)
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot with all correct data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call fetch with the correct url', () =>{
    URL = "https://swapi.co/api/films";
    wrapper.instance().componentDidMount();
    expect(window.fetch).toHaveBeenCalledWith(URL)
  })

  it('should call fetch with the correct url in gatherTypes', () =>{
    URL = "https://swapi.co/api/planets";
    wrapper.instance().gatherTypes(URL);
    expect(window.fetch).toHaveBeenCalledWith(URL)
  })

  it('should change state to the correct type in filterByType', () =>{
    let type = "people";
    wrapper.instance().filterByType(type);
    expect(wrapper.state('currentCategory')).toEqual("people");
  })

  it('should call return a random movie', () =>{
    wrapper.instance().randomMovie("https://swapi.co/api/films")
    expect(wrapper.state().randomMovie).toEqual({})
  })

  it.skip('should call gatherTypes with correct URL in filterByType', () =>{
    let type = "planets";
    wrapper.instance().filterByType(type);
    expect(filterByType).toHaveBeenCalledWith(`https://swapi.co/api/${type}/`)
  })

  it.skip('should call randomMovie on componentDidMount', () =>{
    wrapper.instance().componentDidMount();
    expect(randomMovie).toHaveBeenCalled()
  })

})

