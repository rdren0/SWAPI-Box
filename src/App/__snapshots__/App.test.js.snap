// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with all correct data passed through 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    filterByType={[MockFunction]}
    randomMovie={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          className="header"
        />,
        <Container
          addFavorites={[Function]}
          cards={Array []}
          category=""
          favorites={Array []}
          filterByType={[Function]}
        />,
        <TextScroll
          crawl={Object {}}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "header",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addFavorites": [Function],
          "cards": Array [],
          "category": "",
          "favorites": Array [],
          "filterByType": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "crawl": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            className="header"
          />,
          <Container
            addFavorites={[Function]}
            cards={Array []}
            category=""
            favorites={Array []}
            filterByType={[Function]}
          />,
          <TextScroll
            crawl={Object {}}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "header",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addFavorites": [Function],
            "cards": Array [],
            "category": "",
            "favorites": Array [],
            "filterByType": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "crawl": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
